jerry-core/parser/js/lexer.cpp

```diff
@@ -1294,16 +1290,16 @@ lexer_parse_token (void)
   }
 
   if (c == LIT_CHAR_SLASH
-      && !(sent_token.type == TOK_NAME
-           || sent_token.type == TOK_NULL
-           || sent_token.type == TOK_BOOL
-           || sent_token.type == TOK_CLOSE_BRACE
-           || sent_token.type == TOK_CLOSE_SQUARE
-           || sent_token.type == TOK_CLOSE_PAREN
-           || sent_token.type == TOK_SMALL_INT
-           || sent_token.type == TOK_NUMBER
-           || sent_token.type == TOK_STRING
-           || sent_token.type == TOK_REGEXP))
+      && !(prev_non_lf_token.type == TOK_NAME
+           || prev_non_lf_token.type == TOK_NULL
+           || prev_non_lf_token.type == TOK_BOOL
+           || prev_non_lf_token.type == TOK_CLOSE_BRACE
+           || prev_non_lf_token.type == TOK_CLOSE_SQUARE
+           || prev_non_lf_token.type == TOK_CLOSE_PAREN
+           || prev_non_lf_token.type == TOK_SMALL_INT
+           || prev_non_lf_token.type == TOK_NUMBER
+           || prev_non_lf_token.type == TOK_STRING
+           || prev_non_lf_token.type == TOK_REGEXP))
```

Comment:
`TOK_CLOSE_PAREN` and `TOK_CLOSE_BRACE` could precede regular expression, for example:
```
if (true) /abc/.exec("abc");
if (true) {} /abc/.exec("abc");
```
This case is harder to support, possibly, we could implement it in another patch.